src/Parser/Serialization/FuncSerializer.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Text;
+
+namespace Parser
+{
+    public class FuncSerializer<T> : ISerialisation
```

Comment:
According to my previous comment about using a JSON lib, this wouldn't be needed? E.g. people using JSON.NET have all the extension hooks they need to customize their output and special types handling.